src/lib/addProvide.js

```diff
@@ -14,6 +14,8 @@ function addProvide (component, options) {
     if (originalBeforeCreate) {
       originalBeforeCreate.call(this)
     }
+
+    component.beforeCreate = originalBeforeCreate
```

Comment:
so....that was pretty hard to isolate as being the issue.  Component was reusing the beforeCreate in between tests.  '

This is passing in chrome but not jsdom, will be able to check this out tomorrow :)

Brings me to a thought, when we mount things does it make sense to clone them or give the option for them to be cloned?